src/lib.rs

```diff
@@ -460,41 +478,48 @@ struct CLruNode<K, V> {
     value: V,
 }
 
-/// An LRU cache with constant time operations.
+/// A weighted LRU cache with constant time operations.
+///
+/// Each value in the cache can have a weight that is retrieved using
+/// the provided [`WeightScale`] implementation. The default scale is
+/// [`ZeroWeightScale`] and always return 0. The number of elements that
+/// can be stored in the cache is conditioned by the sum of [`CLruCache::len`]
+/// and [`CLruCache::weight`]:
+///
+/// [`CLruCache::len`] + [`CLruCache::weight`] <= [`CLruCache::capacity`]
```

Comment:
You're conflating two ideas of "capacity" here: capacity in terms of number of entries, and capacity in terms of supporting a total of (user-provided) weights.

It's working, as far as I can see, and it's a nice idea. But maybe explicitly mentioning this conflation (in addition to the formula above, I mean) would be a good idea.

Reply:
I added a more detailed description in the crate level documentation, hopefully that is better now :+1: